src/components/MDXContainer.js

```diff
@@ -23,8 +24,27 @@ const components = {
   Important,
   Tip,
   Intro,
-  code: (props) => <CodeSnippet {...props} />,
+  code: ({
+    className,
+    copy,
+    lineNumbers,
+    live,
+    lineHighlight,
+    preview: _preview,
+    ...props
+  }) => (
+    <CodeBlock
+      copyable={copy !== 'false'}
+      highlightedLines={lineHighlight}
+      language={className?.replace('language-', '')}
+      lineNumbers={lineNumbers === 'true'}
+      live={live === 'true'}
+      {...props}
+    />
+  ),
```

Comment:
Out of curiosity, why did you decide to destructure the props here rather than in the component? Is this so that we don't have to change the props being used in the `.mdx` files?

That would make sense and, if that's the case, we might want to think about how we approach the problem of refactoring code without altering the content. Maybe not here and now, but we might want to add a layer to handle that formatting.

Reply:
I'm not sure I follow what you're asking. Are you asking why I didn't put this logic inside of `<CodeBlock />` itself?

I wanted `<CodeBlock />` to have a more natural API since its used in more contexts than just MDX files. For example, I wanted to use the booleans `true` and `false` instead of their string representations. For the language, I felt it more natural to provide a `language` prop rather than set as a `className`. If I moved the logic inside of `<CodeBlock />`, it would present an awkward API for usages outside of MDX.

To illustrate, here is what the API would look like in something like `<ReferenceExample />` if we moved this logic inside of `<CodeBlock />` itself.

```js
<CodeBlock
  copy="false"
  live="true"
  className="language-jsx"
/>
```

Reply:
I think we're saying the same thing. You're manipulating the props _here_ so that the `<CodeBlock />` component can have a simpler API.

I kinda spaced out on the fact that we're using the `<CodeBlock />` component in MDX _and_ in JSX.

Either way, I _do_ still think we should think about how we can refactor our components (and use our components in JSX) without needing to update how we use them in MDX. A discussion for another time, but something we will probably want to have a plan for.

Reply:
> think about how we can refactor our components (and use our components in JSX)

To clarify, are you referring to other components and not just the `CodeBlock`? The code block is a little bit of a special use case because it is generated using 3 backticks. For other components (like `<Step />`), you can provide props to it like you would in JSX, which means we shouldn't need any special parsing for them:

These should all work in MDX:
```mdx
<Step steps={4} onClick={() => console.log('clicked')}>
  Some content for the step
</Step>
```

Reply:
> I think we're saying the same thing. You're manipulating the props here so that the <CodeBlock /> component can have a simpler API.

And yes that is correct 😄 .

Technically speaking, I'm providing a function component inline here, I just didn't name it before assigning it to the `code` key in the `components` object. I could have just as easily done the following:

```
const MDXCodeBlock = ({ className, copy }) => (
  <CodeBlock copyable={copy === 'true'} language={className.replace('language-', '')} />
)

const components = {
  // ...
  code: MDXCodeBlock
}
```